### ISSUE TITLE
KerasTensor Exception when using DeepFace.extract_faces

### ISSUE THREAD

**Issue by AaronMk44** [serengil/deepface#1100]
Hi, I encountered an error while trying to extract faces from an image using the deepface library.

```
faces = DeepFace.extract_faces(
  img_path=img_path,
  detector_backend = 'retinaface',
  enforce_detection=False,
  align = True
)
```

I got this error:

```
Exception encountered when calling MaxPooling2D.call().\n\n\x1b[1mArgument `padding` must be either 'valid' or 'same'. Received: padding=VALID\x1b[0m\n\nArguments received by MaxPooling2D.call():\n  • args=('<KerasTensor shape=(None, None, None, 64), dtype=float32, sparse=False, name=keras_tensor_5>',)\n  • kwargs=<class 'inspect._empty'>
```

It appears the issue is linked to the Invocation of MaxPooling2D.call(). At present it appears the value being passed for the argument `padding` is VALID in uppercase, when the expected value should be lower case. 

Would it be possible to change this to lowercase so that the error is fixed? or perhaps could you advise how to resolve this issue?

**Comment by AaronMk44** [comment]
It's alright, just found the solution🤦‍♂️

That's the corrected code. The solution was to change the detector_backend to `ssd` from `retinaface`
```
faces = DeepFace.extract_faces(
  img_path=img_path,
  detector_backend = 'ssd',
  enforce_detection=False,
  align = True
)

```

**Comment by AndreaLanfranchi** [comment]
I don't think this solves the problem. You simply changed from a non-working detector to a workin one.

**Comment by serengil** [comment]
@AaronMk44 what is your tf version?

**Comment by AaronMk44** [comment]
@AndreaLanfranchi you're right, it doesn't solve the problem. But at least we all know the issue exists and needs to be worked on

**Comment by AaronMk44** [comment]
@serengil  I'm using:

Metadata-Version: 2.1
Name: tensorflow-intel
Version: 2.16.1

**Comment by serengil** [comment]
can you retry this with downgrading tf? e.g. 2.9.0

**Comment by AaronMk44** [comment]
I'll try it later on, cause changing my tf version right now may break many things completely. 
"If it's working, don't change it" 🥲

Otherwise, thank you for the assistance, it's really appreciated and great work on the library too!

**Comment by serengil** [comment]
99.9% tf version bug, i have been getting similar errors recently a lot. most probably, tf is broken somehow in its new version.

**Comment by serengil** [comment]
most probably, tf will sort this soon

if we put a version control, we will be out dated even if tf updated

**Comment by AaronMk44** [comment]
Oh I see your point, so in case they don't, what would be the most sustainable solution?
